### PR TITLE
Feature/387 document job log

### PR DESCRIPTION
**Pull request by edmundreinhardt**
The job log documentation was done from scratch, including screen caps.

A few menu items were renamed for clarity
`Copy` to `Copy Command`
`Clear Job Log` to `Clear Previous Job Logs`

The `Show Build Output` command was removed as it no longer worked.

The panel was renamed from `IBM i Project Explorer` to `IBM i Job Log`

Context menu items  were added to the lower items on the tree where appropriate
`Copy Command` on compile string
`Filter by Message Severity` on job log
`Copy Command` and `Show Object Build Output` on object

### CODE REVIEW

**Comment by edmundreinhardt** [comment]
@irfanshar You definitely need to review this PR as it might invalidate some of your existing tests.  It will also help you understand the job log viewer design and what other functions can be tested.

@SanjulaGanepola Feel free to review if you have time.

File: docs/pages/jobLog/viewing-job-logs.md [hunk 1]
```diff
@@ -1,5 +1,64 @@
...
+![Job Log panel](joblog.png)
+
+## Job log tree contents
+
+### Project level
+The top level of this tree is the project name with the project description as its details.  This allows the job logs of multiple projects to be shown in this view.  Only the latest build is persisted in the project, but the other jobs are kept in memory until the workspace is restarted or the `Clear Previous Job Logs` action on the project is run.
+
+<img src="action_clear.png" width="300">
+
+The `Show Job Log` action, which is also available inline, will load the backing .json file into the editor.  This might be useful for searching, etc.
```
Comment (SanjulaGanepola) [inline docs/pages/jobLog/viewing-job-logs.md:14]: > load the backing .json

Should be "load the joblog.json"?

File: docs/pages/jobLog/viewing-job-logs.md [hunk 2]
```diff
@@ -1,5 +1,64 @@
...
+The remaining items are the messages that were produced in the job log when executing this command.  While there are no actions on messages, the hover shows all of the message details including second level help.
+
+ <img src="second_level_message.png" width="500">
+
+## Where does the data come from
+Build commands such as BOB (https://github.com/IBM/ibmi-bob) and ARCAD generate a file named `.logs/joblog.json`. They also generate text files in the same directory with names like `target_object.splf` that contain the PASE standard out or spool file equivalent for each command, including the compile listing.  These files are copied to the `.logs` directory in the project root.
+
+<img src="joblog_files.png" width="300">
+
+This copy process is automatic when invoking build or compile action for `IBM i Project Explorer` projects. For `Code for IBM i` actions the downloading of these files can be configured by the `postdownLoad` attribute in the `actions.json`  file.  The `.evfevent` directory specified here is necessary to get the compiler errors in the `Problems` view.
```
Comment (SanjulaGanepola) [inline docs/pages/jobLog/viewing-job-logs.md:61]: > configured by the `postdownLoad` 

Casing should be `postDownload`

**Review by SanjulaGanepola — APPROVED** [review]
Content looks good. Just a couple comments to address.

For images, when running the docs locally (`npm run docs`), most images that you rendered using `img` tags did not show up. To use the `img` tags, I think you need to prefix the path with `/pages/jobLog/`. Also, to be consistent with our existing docs, we typically store all the images in the `assets` folder. If you move the images, the paths will need to be updated accordingly.

image

**Comment by SanjulaGanepola** [comment]
When testing against `bob-recursive-example`, I noticed a 2 bugs actually:

1) Tree item labels were missing. Taking a look at the joblog.json file, some attributes did not seem to be set (a bob issue?):

image

image

2) Second level help not displaying correctly:

image

**Review by irfanshar — APPROVED** [review]

**Comment by irfanshar** [comment]
> @irfanshar You definitely need to review this PR as it might invalidate some of your existing tests. It will also help you understand the job log viewer design and what other functions can be tested.
> 
> @SanjulaGanepola Feel free to review if you have time.

Definitely some of our tests will need to be modified.

File: docs/pages/jobLog/viewing-job-logs.md (hunk 2 above)
Comment (edmundreinhardt) [reply docs/pages/jobLog/viewing-job-logs.md:61]: thank you, done.

File: docs/pages/jobLog/viewing-job-logs.md (hunk 1 above)
Comment (edmundreinhardt) [reply docs/pages/jobLog/viewing-job-logs.md:14]: thank you, done

**Comment by edmundreinhardt** [comment]
> Content looks good. Just a couple comments to address.
> 
> For images, when running the docs locally (`npm run docs`), most images that you rendered using `img` tags did not show up. To use the `img` tags, I think you need to prefix the path with `/pages/jobLog/`. Also, to be consistent with our existing docs, we typically store all the images in the `assets` folder. If you move the images, the paths will need to be updated accordingly.
Thank you, I ended up using img tags because the images were too big otherwise

**Comment by edmundreinhardt** [comment]
> > When testing against `bob-recursive-example`, I noticed 2 bugs actually:
> > 
> > 1. Tree item labels were missing. Taking a look at the joblog.json file, some attributes did not seem to be set (a bob issue?):
> > 
> > image
> > image
> > 
> > 2. Second level help not displaying correctly:
> > 
> > image
> 
> Definitely a bob issue. [IBM/ibmi-bob#306](https://github.com/IBM/ibmi-bob/issues/306)
> 
> For certain commands (for example DLTOBJ), no object is being generated. Curious as what we want to display in the label for this command (since other commands display the created object), and what we want the value of these empty attributes to be in these instances.

This will be fixed with the BOB issue

**Pull request merged**